lib/WeBWorK/FormatRenderedProblem.pm

```diff
@@ -0,0 +1,313 @@
+################################################################################
+# WeBWorK Online Homework Delivery System
+# Copyright &copy; 2000-2022 The WeBWorK Project, https://github.com/openwebwork
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of either: (a) the GNU General Public License as published by the
+# Free Software Foundation; either version 2, or (at your option) any later
+# version, or (b) the "Artistic License" which comes with this package.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See either the GNU General Public License or the
+# Artistic License for more details.
+################################################################################
+
+=head1 NAME
+
+FormatRenderedProblem.pm
+
+=cut
+
+package WeBWorK::FormatRenderedProblem;
+
+use strict;
+use warnings;
+
+use JSON;
+use Digest::SHA qw(sha1_base64);
+use Mojo::Util qw(xml_escape);
+use Mojo::DOM;
+
+use WeBWorK::Localize;
+use WeBWorK::AttemptsTable;
+use WeBWorK::Utils qw(getAssetURL);
+use WeBWorK::Utils::LanguageAndDirection;
+
+sub formatRenderedProblem {
+	my $c = shift;
+	my $rh_result = shift;
+	my $inputs_ref = $rh_result->{inputs_ref};
+
+	my $renderErrorOccurred = 0;
+
+	my $problemText = $rh_result->{text} // '';
+	$problemText   .= $rh_result->{flags}{comment} if ( $rh_result->{flags}{comment} && $inputs_ref->{showComments} );
+
+	if ($rh_result->{flags}{error_flag}) {
+		$rh_result->{problem_result}{score} = 0;    # force score to 0 for such errors.
+		$renderErrorOccurred                = 1;
+	}
+
+	my $SITE_URL        = $inputs_ref->{baseURL} || $main::basehref;
+	my $FORM_ACTION_URL = $inputs_ref->{formURL} || $main::formURL; 
+
+	my $displayMode = $inputs_ref->{displayMode} // 'MathJax';
+
+	# HTML document language setting
+	my $formLanguage = $inputs_ref->{language} // 'en';
+
+	# Third party CSS
+	# The second element of each array in the following is whether or not the file is a theme file.
+	# customize source for bootstrap.css
```

Comment:
The comment about the second element determining if the file is a theme file should be deleted.  It isn't relevant for the standalone renderer.

The second comment seems out of place.